BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt

```diff
@@ -126,44 +115,20 @@ open class BraintreeClient @VisibleForTesting internal constructor(
      * @param clientTokenProvider An implementation of [ClientTokenProvider] that [BraintreeClient]
      * will use to fetch a client token on demand.
      * @param returnUrlScheme     A custom return url to use for browser and app switching
+     * @param appLinkReturnUri TODO
      */
+    @JvmOverloads
     constructor(
         context: Context,
         clientTokenProvider: ClientTokenProvider,
-        returnUrlScheme: String
```

Comment:
The signature of the constructors would be changing, but it wouldn't be a breaking change as the `returnUrlScheme` param goes from non-null to nullable. It's possible that a merchant can now pass in null, but this shouldn't break anything as the internal default value was set to null.

I feel that this greatly reduced the number of constructors we have to manage, along with making it much easier to add additional params like the new `appLinkReturnUri`.

```kotlin
    constructor (
        context: Context,
        authorization: String,
        returnUrlScheme: String
    )
```
would now be
```kotlin
    constructor (
        context: Context,
        authorization: String,
        returnUrlScheme: String?
    )
```

Is there anything that I'm overlooking here? I want to ensure this is a viable change before moving forward with additional app link changes.

Reply:
What does this look like for a Java integration? I think that would be my only point of concern - if they were previously using `BraintreeClient(context, authorization)`, do they now need to pass `(context, authorization, null)` ?

Reply:
The Java integration that uses `BraintreeClient(context, authorization)` would be unchanged. Null does not need to be passed because of the default value in Kotlin.

Here's the decompiled Java code that's generated:

```Java
   @JvmOverloads
   public BraintreeClient(@NotNull Context context, @NotNull String authorization, @Nullable String returnUrlScheme, @Nullable Uri appLinkReturnUri) {
      Intrinsics.checkNotNullParameter(context, "context");
      Intrinsics.checkNotNullParameter(authorization, "authorization");
      Object var5 = null;
      Object var6 = null;
      this(new BraintreeOptions(context, (String)null, returnUrlScheme, appLinkReturnUri, authorization, (ClientTokenProvider)var6, (String)var5, 98, (DefaultConstructorMarker)null));
   }

   @JvmOverloads
   public BraintreeClient(@NotNull Context context, @NotNull String authorization, @Nullable String returnUrlScheme) {
      this(context, (String)authorization, returnUrlScheme, (Uri)null, 8, (DefaultConstructorMarker)null);
   }

   @JvmOverloads
   public BraintreeClient(@NotNull Context context, @NotNull String authorization) {
      this(context, (String)authorization, (String)null, (Uri)null, 12, (DefaultConstructorMarker)null);
   }
```

The only gotcha for Java integrations is if a merchant wants to pass in an `appLinkReturnUri`, null must be passed in for the `returnUrlScheme` param. It doesn't look like `BraintreeClient(@NotNull Context context, @NotNull String authorization, @Nullable Uri appLinkReturnUri)` is generated. Would this be okay?

The only other solution I see is to double the number of public constructors which would take us from the existing 4 to 8 😟

Reply:
That seems ok to me 👍

Reply:
Yep, I agree! I believe the `BraintreeClient` will be hidden from the merchant, so we'll have to add these URLs to the respective request objects for v5.